docs/scos/dev/tutorials-and-howtos/howtos/howto-reduce-jenkins-execution-costs-without-refactoring.md

```diff
@@ -0,0 +1,483 @@
+use Spryker\Zed\Queue\Business\Worker\WorkerInterface;
+use Spryker\Zed\Queue\Communication\Console\QueueWorkerConsole;
+
+class NewWorker implements WorkerInterface
+{
+    /**
+     * @var \Spryker\Zed\Queue\Business\Process\ProcessManagerInterface
+     */
+    protected ProcessManagerInterface $processManager;
+
+    /**
+     * @var \Pyz\Zed\Queue\QueueConfig
+     */
+    protected QueueConfig $queueConfig;
+
+    /**
+     * @var \Spryker\Client\Queue\QueueClientInterface
+     */
+    protected QueueClientInterface $queueClient;
+
+    /**
+     * @var array<string>
+     */
+    protected array $queueNames;
+
+    /**
+     * @var \Spryker\Zed\Queue\Business\SignalHandler\SignalDispatcherInterface
+     */
+    protected SignalDispatcherInterface $signalDispatcher;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected LoggerInterface $logger;
+
+    /**
+     * @var \Pyz\Zed\Queue\Business\Strategy\QueueProcessingStrategyInterface
+     */
+    protected QueueProcessingStrategyInterface $queueProcessingStrategy;
+
+    /**
+     * @var \SplFixedArray<\Symfony\Component\Process\Process>
+     */
+    protected SplFixedArray $processes;
+
+    /**
+     * @var int
+     */
+    private int $runningProcessesCount = 0;
+
+    /**
+     * @var \Pyz\Zed\Queue\Business\Worker\WorkerStats
+     */
+    private WorkerStats $stats;
+
+    /**
+     * @var \Pyz\Zed\Queue\Business\SystemResources\SystemResourcesManagerInterface
+     */
+    private SystemResourcesManagerInterface $sysResManager;
+
+    /**
+     * @var array<string, float>
+     */
+    private array $timers = [];
+
+    public function __construct(
+        ProcessManagerInterface $processManager,
+        QueueConfig $queueConfig,
+        QueueClientInterface $queueClient,
+        array $queueNames,
+        QueueProcessingStrategyInterface $queueProcessingStrategy,
+        SignalDispatcherInterface $signalDispatcher,
+        SystemResourcesManagerInterface $sysResManager,
+        LoggerInterface $logger
+    ) {
+        $this->processManager = $processManager;
+        $this->queueConfig = $queueConfig;
+        $this->queueClient = $queueClient;
+        $this->queueNames = $queueNames;
+        $this->queueProcessingStrategy = $queueProcessingStrategy;
+        $this->signalDispatcher = $signalDispatcher;
+        $this->signalDispatcher->dispatch($this->queueConfig->getSignalsForGracefulWorkerShutdown());
+        $this->sysResManager = $sysResManager;
+        $this->logger = $logger;
+
+        $this->processes = new SplFixedArray($this->queueConfig->getQueueWorkerMaxProcesses());
+        $this->stats = new WorkerStats();
+    }
+
+    /**
+     * @param string $timerName
+     * @param string $message
+     * @param string $level
+     * @param int $intervalSec
+     *
+     * @return void
+     */
+    protected function logNotOftenThan(string $timerName, string $message, string $level = 'debug', int $intervalSec = 1): void
```

Comment:
mb we can strip this example from additional code like logging and just replace those with comments e.g. //Here you can log smth here